cmd/kpod/run.go

```diff
@@ -19,35 +21,86 @@ var runCommand = cli.Command{
 }
 
 func runCmd(c *cli.Context) error {
-	if len(c.Args()) != 1 {
-		return errors.Errorf("must specify name of image to create from")
-	}
 	if err := validateFlags(c, createFlags); err != nil {
 		return err
 	}
-	runtime, err := getRuntime(c)
+	runtime, err := libpod.NewRuntime()
 	if err != nil {
 		return errors.Wrapf(err, "error creating libpod runtime")
 	}
 
-	createConfig, err := parseCreateOpts(c)
+	createConfig, err := parseCreateOpts(c, runtime)
 	if err != nil {
 		return err
 	}
 
+	createImage := runtime.NewImage(createConfig.image)
+
+	if !createImage.HasImageLocal() {
+		// The image wasnt found by the user input'd name or its fqname
+		// Pull the image
+		fmt.Printf("Trying to pull %s...", createImage.PullName)
+		createImage.Pull()
+	}
+
 	runtimeSpec, err := createConfigToOCISpec(createConfig)
+	logrus.Debug("spec is ", runtimeSpec)
+	if err != nil {
+		return err
+	}
+
+	imageName, err := createImage.GetFQName()
+	if err != nil {
+		return err
+	}
+	logrus.Debug("imageName is ", imageName)
+
+	imageID, err := createImage.GetImageID()
 	if err != nil {
 		return err
 	}
+	logrus.Debug("imageID is ", imageID)
+
+	options, err := createConfig.GetContainerCreateOptions(c)
+	if err != nil {
+		return errors.Wrapf(err, "unable to parse new container options")
+	}
 
-	ctr, err := runtime.NewContainer(runtimeSpec)
+	// Gather up the options for NewContainer which consist of With... funcs
+	options = append(options, libpod.WithRootFSFromImage(imageID, imageName, false))
+	ctr, err := runtime.NewContainer(runtimeSpec, options...)
 	if err != nil {
 		return err
 	}
 
-	// Should we also call ctr.Create() to make the container in runc?
+	logrus.Debug("new container created ", ctr.ID())
+	if err := ctr.Create(); err != nil {
+		return err
+	}
+	logrus.Debug("container storage created for ", ctr.ID())
 
-	fmt.Printf("%s\n", ctr.ID())
+	if c.String("cid-file") != "" {
+		libpod.WriteFile(ctr.ID(), c.String("cid-file"))
+		return nil
+	}
+	// Start the container
+	if err := ctr.Start(); err != nil {
+		return errors.Wrapf(err, "unable to start container ", ctr.ID())
+	}
+	logrus.Debug("started container ", ctr.ID())
+	if createConfig.tty {
+		// Attach to the running container
+		keys := ""
+		if c.String("detach-keys") != "" {
+			keys = c.String("detach-keys")
+		}
+		logrus.Debug("trying to attach to the container %s", ctr.ID())
+		if err := ctr.Attach(false, keys); err != nil {
```

Comment:
Just pass in c.String("detach-keys")

Reply:
Also, we should only Attach() if -d is not passed. If -d is passed, even if we have -t -i, we should just print the container ID and exit.

Reply:
Yup this is how docker works.